subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java

```diff
@@ -53,6 +57,7 @@ public BuildCacheServiceProvider(BuildCacheConfigurationInternal buildCacheConfi
 
     public BuildCacheService createBuildCacheService() {
         if (!startParameter.isBuildCacheEnabled()) {
+            buildOperationExecutor.run(new FinalizeBuildCacheConfigurationBuildOperation(null, null));
```

Comment:
would it be useful to collect the build cache information for builds that have the build cache disabled? e.g., someone normally uses the build cache, but then runs a single build with `--no-build-cache`?